src/frequenz/sdk/timeseries/battery_pool/_component_metric_fetcher.py

```diff
@@ -0,0 +1,305 @@
+        """
+
+        try:
+            data = await self._receiver.receive()
+        except ChannelClosedError as err:
+            _logger.error(
+                "battery_data channel for component %d was closed. Err: %s",
+                self._component_id,
+                str(err),
+            )
+            return None
+
+        metrics = {}
+        for mid in self._metrics:
+            metrics[mid] = _BatteryDataMethods[mid](data)
+        return ComponentMetricsData(self._component_id, data.timestamp, metrics)
+
+
+class LatestInverterMetricsFetcher(ComponentMetricFetcher):
+    """Subscribe for the latest inverter data using MicrogridApiClient."""
+
+    def __init__(
+        self,
+        component_id: int,
+        metrics: List[ComponentMetricId],
+    ) -> None:
+        """Create class instance.
+
+        Args:
+            component_id: Component id
+            metrics: Metrics needed for the given component
+
+        Raises:
+            ValueError: If any metrics are not supported.
+        """
+        super().__init__(component_id, metrics)
+        if any(map(lambda mid: mid not in _InverterDataMethods, metrics)):
+            raise ValueError(
+                "Some battery metrics are not supported, supported: "
+                f"{str(_InverterDataMethods.keys())}, requested: {str(metrics)}"
+            )
+
+    async def subscribe(self) -> None:
+        """Subscribe for this component metrics."""
+        api = get_microgrid().api_client
+        # Maxsize of the receivers should be 1 to make sure we get only
+        # the latest message.
+        self._receiver = await api.inverter_data(self._component_id, maxsize=1)
+
+    async def fetch_next(self) -> Optional[ComponentMetricsData]:
+        """Fetch the component metrics.
+
+        User should subscribe for the component first.
+
+        Returns:
+            Component metrics data.
+        """
+        try:
+            data = await self._receiver.receive()
+        except ChannelClosedError as err:
+            _logger.error(
+                "inverter_data channel for component %d was closed. Err: %s",
+                self._component_id,
+                str(err),
+            )
+            return None
+
+        metrics = {}
+        for mid in self._metrics:
+            metrics[mid] = _InverterDataMethods[mid](data)
+        return ComponentMetricsData(self._component_id, data.timestamp, metrics)
+
+
+@dataclass
+class ResamplerArgs:
+    """Arguments needed to subscribe for the resampling actor."""
+
+    namespace: str
+    """The unique namespace to allow reuse of streams in the data pipeline."""
+
+    resampler_subscription_sender: Sender[ComponentMetricRequest]
+    """Sender channel to send the metric subscription request."""
+
+    channel_registry: ChannelRegistry
+    """Channel registry instance to get the receiver for the resampling channel."""
+
+
+class ResampledComponentMetricFetcher(ComponentMetricFetcher):
```

Comment:
This is somehow analogous to the `LogicalMeter`, right?

Reply:
Yes, but was formula was computed for  working  components only.
In the end it turns out we need to compute active_power for all batteries in pool. (independently if they are working or not).
So I removed this class and I will use FormulaEngine for active power in future (next PR)